File: solutions/python/triangle/1/triangle.py
def not_degenerate(sides: list[int]) -> bool:
    [small, middle, large] = sorted(sides)
    return small > 0 and  large < (middle + small)

def equilateral(sides: list[int]) -> bool:
    return not_degenerate(sides) and sides[0] == sides[1] and sides[1] == sides[2]

def isosceles(sides: list[int]) -> bool:
    return not_degenerate(sides) and len(set(sides)) < 3

def scalene(sides: list[int]) -> bool:
    return not_degenerate(sides) and sides[0] != sides[1] and sides[1] != sides[2] and sides[0] != sides[2]

File: solutions/php/triangle/1/Triangle.php
<?php

declare(strict_types=1);

class Triangle
{
    private $sides = [];

    public function __construct(float $a, float $b, float $c)
    {
        if ($a <= 0 || $b <= 0 || $c <= 0) throw new \Exception();

        $this->sides = [$a, $b, $c];
        rsort($this->sides);

        if ($this->sides[0] > ($this->sides[1] + $this->sides[2])) throw new \Exception();
    }

    public function kind(): string
    {
        if ($this->sides[0] == $this->sides[1] && $this->sides[1] == $this->sides[2]) {
            return 'equilateral';
        } else if ($this->sides[0] != $this->sides[1] && $this->sides[1] != $this->sides[2] && $this->sides[0] != $this->sides[2]){
            return 'scalene';
        } else {
            return 'isosceles';
        }
    }
}
